src/states/elder/mod.rs

```diff
@@ -710,7 +727,91 @@ impl Elder {
             return;
         }
 
-        self.vote_for_event(AccumulatingEvent::Online(pub_id));
+        // This joining node is being relocated to us.
+        let age = if let Some(payload) = relocate_payload {
+            if !payload.verify_identity(&pub_id) {
+                debug!(
+                    "{} - Ignoring relocation JoinRequest from {} - invalid signature.",
+                    self, pub_id
+                );
+                return;
+            }
+
+            let details = payload.details;
+
+            if !self
+                .chain
+                .our_prefix()
+                .matches(&details.content().destination)
+            {
+                debug!("{} - Ignoring relocation JoinRequest from {} - destination {} doesn't match our prefix {:?}.",
+                    self, pub_id, details.content().destination, self.chain.our_prefix());
+                return;
+            }
+
+            let age = details.content().age;
+            let message = SignedRoutingMessage::from(details);
+
+            if let Err(err) = message.check_integrity() {
```

Comment:
As the message SingedRoutingMessage, can we use the new `check_signed_message_integrity` and `check_signed_message_trust` function directly?

Reply:
It wouldn't buy as much. Also I wanted the log messages in case of failure to be different here.

Reply:
Ok. If you think a different log message is important here.  :)